core/domain/classifier_services.py

```diff
@@ -224,3 +226,117 @@ def delete_classifier(classifier_id):
+        classifier_training_job_model.status,
+        classifier_training_job_model.training_data)
+
+def get_classifier_training_job_by_id(job_id):
+    """Gets a classifier training job from a job_id.
+
+    Args:
+        job_id: str. ID of the classifier training job.
+
+    Returns:
+        classifier_training_job: Domain object for the classifier training job.
+
+    Raises:
+        Exception: Entity for class ClassifierTrainingJobModel with id not
+            found.
+    """
+    classifier_training_job_model = (
+        classifier_models.ClassifierTrainingJobModel.get(job_id))
+    classifier_training_job = get_classifier_training_job_from_model(
+        classifier_training_job_model)
+    return classifier_training_job
+
+
+def _create_classifier_training_job(classifier_training_job):
+    """Creates classifier training job model in the datastore given a
+       classifier training job domain object.
+
+    Args:
+        classifier_training_job: Domain object for the classifier training job.
+
+    Returns:
+        job_id: str. ID of the classifier training job.
+    """
+    job_id = classifier_models.ClassifierTrainingJobModel.create(
+        classifier_training_job.algorithm_id, classifier_training_job.exp_id,
+        classifier_training_job.exp_version_when_created,
+        classifier_training_job.training_data,
+        classifier_training_job.state_name)
+    return job_id
+
+
+def _update_classifier_training_job(classifier_training_job_model, state_name,
+                                    status):
+    """Updates classifier training job model in the datastore given a
+       classifier training job domain object.
+
+    Args:
+        classifier_training_job_model: Classifier training job model instance
+            in datastore.
+        state_name: The name of the state.
+        status: The status of the job.
+
+    Note: All of the proeprties of a classifier training job are immutable,
+        except for state_name and status.
+    """
+    classifier_training_job_model.state_name = state_name
+    classifier_training_job_model.status = status
+    classifier_training_job_model.put()
+
+
+def save_classifier_training_job(classifier_training_job):
+    """Checks for the existence of the model.
+    If the model exists, it is updated using _update_classifier_training_job
+        method.
+    If the model doesn't exist, it is created using
+        _create_classifier_training_job method.
+
+    Args:
+        classifier_training_job: Domain object for the classifier training job.
+
+    Returns:
+        job_id: str. ID of the classifier training job.
+    """
+    job_id = classifier_training_job.job_id
+    classifier_training_job_model = (
+        classifier_models.ClassifierTrainingJobModel.get(job_id, False))
+    classifier_training_job.validate()
+    if classifier_training_job_model is None:
+        job_id = _create_classifier_training_job(classifier_training_job)
```

Comment:
Weird. I thought @anmolshkl left a comment here, but I can't find it.

@pranavsid98 I think having a "dummy job id" rings alarm bells. However I'm finding it hard to advise on the correct design because I don't understand the context in which this function will be called. I think building out the end-to-end pipeline (including the controllers) would make it clearer what functionality is needed at this layer. Do you think you could add that to this PR, as well as tests for the "classifier already exists" and "classifier does not exist" cases?

Basically, like @prasanna08, I'm wondering where the classifier_training_job that seems to be getting inputted comes from.